client/landing/gutenboarding/onboarding-block/vertical-background.tsx

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent, useState } from 'react';
+import { SwitchTransition, CSSTransition } from 'react-transition-group';
+
+export interface VerticalBackgroundProps {
+	id: string;
+	onLoad: () => void;
+}
+
+const VerticalBackground: FunctionComponent< VerticalBackgroundProps > = ( { id, onLoad } ) => {
+	const [ imageId, setimageId ] = useState( '' );
+
+	return (
+		<>
+			<img
+				style={ { display: 'none' } }
+				src={ `/calypso/images/verticals/${ id }.jpg` }
```

Comment:
I'm getting 404s for all the verticals for which we don't have images yet. It'd be good to avoid that :thinking:

Reply:
BTW I assume this `<img />` is for preloading? Would be good to add a little comment to document that :slightly_smiling_face:

Reply:
It took me a while to fully grok `id` prop and `imageId` state handling. Maybe it'd help to change their names a bit. I think the differentiatior shouldn't be `image` but maybe rather than one of the is the currently shown image? I.e. maybe `imageId` -> `currentId`?

Reply:
BTW here's an alternative approach to image (pre)loading (and error handling), using a Webpack loader:

https://github.com/Automattic/wp-calypso/blob/7c7dcd6fdd0f0ee2cd633e25677b99274e55175d/client/lib/flags/index.js

Used e.g. here: https://github.com/Automattic/wp-calypso/blob/master/client/components/phone-input/country-flag.jsx

Reply:
> BTW I assume this `<img />` is for preloading? Would be good to add a little comment to document that 🙂

Added comment regarding this decision only here on GH. Good point to highlight that also in code.
https://github.com/Automattic/wp-calypso/pull/38221/files#r357145183